src/chunk.c

```diff
@@ -3214,39 +3215,30 @@ ts_chunk_drop_process_materialization(Oid hypertable_relid,
  * only if a user explicitly specifies the table name
  */
 List *
-ts_chunk_do_drop_chunks(Oid table_relid, Datum older_than_datum, Datum newer_than_datum,
+ts_chunk_do_drop_chunks(Hypertable *ht, Datum older_than_datum, Datum newer_than_datum,
 						Oid older_than_type, Oid newer_than_type,
 						CascadeToMaterializationOption cascades_to_materializations,
-						int32 log_level, bool user_supplied_table_name, List **affected_data_nodes)
+						int32 log_level, List **affected_data_nodes)
 
 {
 	uint64 i = 0;
 	uint64 num_chunks = 0;
 	Chunk *chunks;
 	List *dropped_chunk_names = NIL;
 	const char *schema_name, *table_name;
-	int32 hypertable_id = ts_hypertable_relid_to_id(table_relid);
+	const int32 hypertable_id = ht->fd.id;
 	bool has_continuous_aggs;
 	List *data_nodes = NIL;
 
-	Assert(OidIsValid(table_relid));
-
-	ts_hypertable_permissions_check(table_relid, GetUserId());
+	ts_hypertable_permissions_check(ht->main_table_relid, GetUserId());
+	lock_referenced_tables(ht->main_table_relid);
```

Comment:
I have been asked to move the comment around already several times, I am not moving it again. It is trivial to go to the function and read the comment there.

Reply:
It is not trivial, since it is deep inside implementation. Furthermore, the function might be shared with other code paths in future, where the comment will not be valid anymore. 
I do write comment, because it is important, and not just for fun. So I don't understand your ignorance.

It is important to describe here why lock is needed. It is not necessary to have long description.

Reply:
Comment moved.